sound/soc/sof/ipc4-pcm.c

```diff
@@ -292,16 +296,29 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 		return -EINVAL;
 	}
 
+	/* NOTE: For the time being chain dma is only used for hdmi
+	 *       audio, which is playback only. If chain dma is ever
+	 *       used for recording, then extracting the pipeline from
+	 *       rtd may need some more lines.
+	 */
```

Comment:
we should absolutely try the capture with the HDaudio external codec. This would be a really good tool to debug.

We should even have a mode where we override the topology and force the pipelines to be dma-chained with a bit set in sof_debug.

Reply:
Now that I think about it, the pcms are symmetrical, no? So in such a case the playback substream should have the use_chain_dma property too, shouldn't it? Or should I check if either direction has the property and get rid of the mask in both cases? Or do we need something more complex?